homeassistant/components/shelly/event.py

```diff
@@ -67,21 +94,80 @@ async def async_setup_entry(
                 async_remove_shelly_entity(hass, EVENT_DOMAIN, unique_id)
             else:
                 entities.append(ShellyRpcEvent(coordinator, key, RPC_EVENT))
+    else:
+        coordinator = get_entry_data(hass)[config_entry.entry_id].block
+        if TYPE_CHECKING:
+            assert coordinator
+            assert coordinator.device.blocks
+
+        for block in coordinator.device.blocks:
+            if (
+                "inputEvent" not in block.sensor_ids
+                or "inputEventCnt" not in block.sensor_ids
+            ):
+                continue
+
+            if BLOCK_EVENT.removal_condition and BLOCK_EVENT.removal_condition(
+                coordinator.device.settings, block
+            ):
+                channel = int(block.channel or 0) + 1
+                unique_id = f"{coordinator.mac}-{block.type}-{channel}"
```

Comment:
```suggestion
                unique_id = f"{coordinator.mac}-{block.description}-{channel}"
```
I noticed during testing that this should be `description` and not `type`